### ISSUE TITLE
Increase timeout for very slow SMTP servers

### ISSUE THREAD

**Issue by mattmartincouk** [atech/postal#295]
We've just had an incident where one customer was getting the same email from us many times but postal said it had failed.

> SOFT FAIL
> Temporary SMTP delivery error when sending to easymx2.easily.co.uk (185.26.230.91)
> Net::ReadTimeout
> Time: 105.99s
> Support Ref: QIZB72LM

On testing that via telnet 185.26.230.91 25, the server is very slow to respond, but does.

However, the message is sent, but postal reports fail and then it is sent again. He got it nine times before he got in touch and we removed it.

Probably not a common thing but one that may need some consideration.

**Comment by willpower232** [comment]
Do you have any log entries for the connections that timed out? Would be interesting to see if postal is already telling us what happened. Presumably the easily system was just taking too long to say OK to postal...

**Comment by mattmartincouk** [comment]
Went to have a look at the log files and seems we've nothing since May so rules out trying to find out what happened there.

Logging is enabled in config too. Interesting (and annoying).

**Comment by mattmartincouk** [comment]
The only logging we have is that in postal but it gives no further clues and the entries appear identical to those in the GUI.

```
^[[35m[worker.8:16931]^[[0m [2017-08-05T08:07:42.493] INFO -- : [QIZB72LM] Connected to 185.26.230.91:25 (easymx2.easily.co.uk)
^[[35m[worker.8:16931]^[[0m [2017-08-05T08:07:42.495] INFO -- : [QIZB72LM] Sending message 9::1782 to x@x
^[[35m[worker.8:16931]^[[0m [2017-08-05T08:08:42.953] INFO -- : [QIZB72LM] Net::ReadTimeout: Net::ReadTimeout
```

Not much use it seems :(

**Comment by willpower232** [comment]
Thats enough to dig a little though. The timeout occurred after a whole minute (which is also the default specified by ruby).

Do you know how long you had to wait for a response when you used telnet?

**Comment by mattmartincouk** [comment]
```
Trying 185.26.230.91...
Connected to easymx2.easily.co.uk.
Escape character is '^]'.
**15 seconds**
220 svcemlmxl18.netnames.net ESMTP ready at Sat, 05 Aug 2017 18:00:13 +0100
helo test.test.com
250 svcemlmxl18.netnames.net Hello x.x [x.x.x.x]
mail from: <>
250 OK
**3 seconds**
rcpt to: x@x
**18 seconds**
250 Accepted
data
354 Enter message, ending with "." on a line by itself
.
**23 seconds**
550 Message content rejected by recipients spam filter.
```
That's 59 seconds but suspect a real request takes longer that isn't rejected. Either way, on my test, it comes a bit too close.

**Comment by willpower232** [comment]
Its probably not a massive job to add in a read timeout config option or just to specify 5 minutes or something in the code, boils down to one line in https://github.com/atech/postal/blob/master/lib/postal/smtp_sender.rb

Have you been able to email this particular person before?

**Comment by mattmartincouk** [comment]
Yes, no issues before. I think their SMTP servers are just massively loaded and thus the time taken is just tripping it slightly.

Given how slow it is, like you say, it's probably sent the final SMTP command for the message, their end has not returned the message to say received and postal has already classed it as timed out and thus tries again and the problem happens again.

**Comment by willpower232** [comment]
A more complete solution would be to flag the message as read timeout and when sending it again, increase the timeout value I guess. Wonder what atechs opinion would be...
